kotlin-analysis/src/main/kotlin/org/jetbrains/dokka/analysis/DRIFactory.kt

```diff
@@ -33,9 +34,10 @@ fun DRI.Companion.from(psi: PsiElement) = psi.parentsWithSelf.run {
         .toList() // We only want exact PsiClass types, not PsiTypeParameter subtype
     DRI(
         packageName = classes.lastOrNull()?.qualifiedName?.substringBeforeLast('.', "") ?: "",
-        classNames = classes.toList().takeIf { it.isNotEmpty() }?.asReversed()?.mapNotNull { it.name }
+        classNames = (if (psi is PsiEnumConstant) listOfNotNull(psiField?.name) + classes.toList().mapNotNull { it.name } else classes.toList().mapNotNull { it.name })
+            .takeIf { it.isNotEmpty() }?.asReversed()
             ?.joinToString("."),
-        callable = psiMethod?.let { Callable.from(it) } ?: psiField?.let { Callable.from(it) },
+        callable = psiMethod?.let { Callable.from(it) } ?: psiField?.takeUnless { psi is PsiEnumConstant }?.let { Callable.from(it) },
```

Comment:
The reason for `takeUnless { psi is PsiEnumConstant }` is not very obvious without context. I think I heard from Vadim that kotlin enums don't have callables? Should we comment this?

Reply:
Added

Reply:
Nice comment, thanks!